docs/deployment/azure/application-insights.md

```diff
@@ -9,60 +9,155 @@ ms.topic: how-to
+builder.AddProject<Projects.Web>("webfrontend")
+    .WithReference(apiService)
+    .WithReference(insights);
 
 builder.Build().Run();
 ```
 
-The preceding reads the connection string from configuration, as this is a secret we should store the value in [app secrets](/aspnet/core/security/app-secrets). Right click on the AppHost project and choose Manage Secrets from the context menu to open the secrets file for the AppHost project. In the file add the key and your specific connection string, the example below is for illustration purposes.
+When you debug the application, .NET Aspire provisions the Application Insights resource the first time it sees the app host resource definition. The Application Insights resource will feed the connection information to each project that references it. This can be seen as an `APPLICATIONINSIGHTS_CONNECTION_STRING` environment variable in the details view for the resource in the .NET Aspire dashboard.
+
+:::image type="content" source="../media/dashboard-with-connection-string.png" lightbox="../media/dashboard-with-connection-string.png" alt-text="Connection string environment variables in the .NET Aspire dashboard":::
+
+Settings for the provisioned Application Insights resource are persisted in the app secrets file.
+
+### Provisioning Application insights during Azure deployment
+
+Using the Application Insights resource above will cause an Application Insights resource to be provisioned when the application is deployed using the Azure Developer CLI (AZD).
+
+Follow the steps in [Deploy a .NET Aspire app to Azure Container Apps using the Azure Developer CLI (in-depth guide)](./aca-deployment-azd-in-depth.md) to deploy the application to Azure Container Apps. AZD will create an Application Insights resource as part of the same resource group, and configure the connection string for each container.
+
+### Manual provisioning of Application Insights resource
+
+Application Insights uses a connection string to tell the OpenTelemetry exporter where to send the telemetry data. The connection string is specific to the instance of Application Insights you want to send the telemetry to. It can be found in the Overview page for the application insights instance.
+
+:::image type="content" source="../media/app-insights-connection-string.png" lightbox="../media/app-insights-connection-string.png" alt-text="Connection string placement in the Azure Application Insights portal UI.":::
+
+If you wish to use an instance of Application Insights that you have provisioned manually, then you should use the `AddConnectioString` API in the app host project to tell the projects/containers where to send the telemetry data. The Azure Monitor distro expects the environment variable to be `APPLICATIONINSIGHTS_CONNECTION_STRING`, so that needs to be explicitly set when defining the connection string.
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+builder.AddAzureProvisioning();
```

Comment:
I think we want to hold off on putting this package in the docs until preview 5. The experience will be much better then @samsp-msft

Reply:
Now you tell me. So what is there for preview 4?